### PR TITLE
fixed mine route issue

### PR DESCRIPTION
**Pull request by kshitij430**
This PR fixes issue https://github.com/Real-Dev-Squad/website-status/issues/375

There were two issues .
1) The Mine component was re evaluating in a loop as the useEffect hook had "response" as  dependency.
2)  The if condition to check if there is a response with length had a not condition which did the opposite. Ex - for 0 results found !0 would evaluate to 1 and !1 would evaluate to a falsy value due to which the setTasks state updating function was not getting executed.

Fixes I have done -
1) Removed response as a dependency as we only want to call the API once when the component is mounted.
2) Removed the condition to check response results as these checks are already done in the JSX return. Ex-> display no results found if length is 0 and otherwise.

### CODE REVIEW

**Comment by kshitij430** [comment]
> Please add description to the PR related to why it was happening and how it was fixed

Already Done. Can you please review it?

**Review by Maheima — CHANGES_REQUESTED** [review]

File: .env.development [hunk 1]
```diff
@@ -1,4 +1,4 @@
-NEXT_PUBLIC_BASE_URL='https://staging-api.realdevsquad.com'
+NEXT_PUBLIC_BASE_URL='https://api.realdevsquad.com'
```
[the remark this answers is not in the document]
Comment (kshitij430) [reply .env.development]: Done. :(

**Review by ivinayakg — DISMISSED** [review]
perfect

**Review by vvaibhavdesai — DISMISSED** [review]

**Review by heyrandhir — DISMISSED** [review]

File: src/pages/mine.tsx [hunk 2]
```diff
@@ -32,11 +32,11 @@ const Mine: FC = () => {
   const { isLoggedIn, isLoading: isAuthenticating } = useAuthenticated();
 
   useEffect(() => {
-    if (isLoggedIn && !Object.keys(response).length) {
+    if (isLoggedIn) {
```
Comment (RitikJaiswal75) [inline src/pages/mine.tsx:35]: since you removed length from the if condition, how are you planning to handle users with 0 tasks?

Comment (RitikJaiswal75) [reply src/pages/mine.tsx:35]: Also please update the tests in case they were supposed to

Comment (harshith-venkatesh) [reply src/pages/mine.tsx:35]: @RitikJaiswal75 , @kshitij430 has added description addressing it

Comment (harshith-venkatesh) [reply src/pages/mine.tsx:35]: Test cases would have helped here

**Comment by harshith-venkatesh** [comment]
Could you add video to showcase the issue being resolved in different scenarios please @kshitij430

File: src/pages/mine.tsx (hunk 2 above)
Comment (Maheima) [reply src/pages/mine.tsx:35]: Can you point us to a resource that can help in writing test for this scenario?

**Review by Maheima — DISMISSED** [review]

**Comment by harshith-venkatesh** [comment]
From contract , response can't be null or undefined or some error?
If we have proper error handling ,we are good with merging

**Comment by kshitij430** [comment]
> From contract , response can't be null or undefined or some error? If we have proper error handling ,we are good with merging

I doubt the API will give a null or undefined but In cases if the response might be null or undefined. It is better to check if there is a required response present and then update the state for tasks.
In a scenario if there is an error the jsx return will handle it.

**Review by Maheima — APPROVED** [review]
LGTM!

**Pull request merged**